apps/server/src/modules/tool/uc/lti-tool.uc.ts

```diff
@@ -0,0 +1,59 @@
+import { LtiToolRepo } from '@shared/repo';
+import { LtiToolDO } from '@shared/domain/domainobject/ltitool.do';
+import { ICurrentUser, IFindOptions, Page, Permission, User } from '@shared/domain';
+import { AuthorizationService } from '@src/modules/authorization';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class LtiToolUc {
+	constructor(private readonly ltiToolRepo: LtiToolRepo, private readonly authorizationService: AuthorizationService) {}
+
+	async findLtiTool(
+		currentUser: ICurrentUser,
+		query: Partial<LtiToolDO>,
+		options: IFindOptions<LtiToolDO>
+	): Promise<Page<LtiToolDO>> {
+		const user: User = await this.authorizationService.getUserWithPermissions(currentUser.userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_VIEW]);
+
+		const tool: Promise<Page<LtiToolDO>> = this.ltiToolRepo.find(query, options);
+		return tool;
+	}
+
+	async getLtiTool(currentUser: ICurrentUser, toolId: string): Promise<LtiToolDO> {
+		const user: User = await this.authorizationService.getUserWithPermissions(currentUser.userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_VIEW]);
+
+		const tool: Promise<LtiToolDO> = this.ltiToolRepo.findById(toolId);
+		return tool;
+	}
+
+	async createLtiTool(currentUser: ICurrentUser, tool: LtiToolDO): Promise<LtiToolDO> {
+		const user: User = await this.authorizationService.getUserWithPermissions(currentUser.userId);
+		this.authorizationService.checkAllPermissions(user, [Permission.TOOL_CREATE]);
+
+		const savedTool: Promise<LtiToolDO> = this.ltiToolRepo.save(tool);
```

Comment:
I think the feathers function has a hook that adds the secret to a tool that gets added to a course, is that being done somehow as well?